### PR TITLE
Oop principles / it-academy.by / JC1-110-22 / Alena Trafimovich/ Done tasks 1-3

### PR DESCRIPTION
**Pull request by AlenaTrafimovich**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/CatHome/Siamese.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.rakovets.course.java.core.practice.oop_principles.CatHome;
+
+public class Siamese extends Cat {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/CatHome/Siamese.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/CatHomeTest/CatTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package com.rakovets.course.java.core.practice.oop_principles.CatHomeTest;
+
+import com.rakovets.course.java.core.practice.oop_principles.CatHome.Cat;
+import com.rakovets.course.java.core.practice.oop_principles.CatHome.Person;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CatTest {
+    private final Cat cat = new Cat("Cat");
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/CatHomeTest/CatTest.java:10]: move to `@BeforeAll`

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Archer.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public class Archer extends Hero {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Archer.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/TypeHero.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public enum TypeHero {
+    ARCHER,
+    MAG,
+    WARRIOR
+}
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/TypeHero.java:7]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/TypeEnemy.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public enum TypeEnemy {
+    VAMPIRE,
+    WITCH,
+    ZOMBIE
+}
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/TypeEnemy.java:7]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Vampire.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public class Vampire extends Enemy {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Vampire.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Warrior.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public class Warrior extends Hero {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Warrior.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Witch.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public class Witch extends Enemy {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Witch.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Zombie.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public class Zombie extends Enemy {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Zombie.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Circle.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
+package com.rakovets.course.java.core.practice.oop_principles.Geometry;
+
+import com.rakovets.course.java.core.util.NumberUtil;
+
+public class Circle extends Ellipse {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Circle.java:6]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Diagonal.java [hunk 11]
```diff
@@ -0,0 +1,6 @@
+package com.rakovets.course.java.core.practice.oop_principles.Geometry;
+
+public interface Diagonal {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Diagonal.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/EquilateralTriangle.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
+package com.rakovets.course.java.core.practice.oop_principles.Geometry;
+
+import com.rakovets.course.java.core.util.NumberUtil;
+
+public class EquilateralTriangle extends Triangle {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/EquilateralTriangle.java:6]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Figure.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
+package com.rakovets.course.java.core.practice.oop_principles.Geometry;
+
+public interface Figure {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Figure.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Foursquare.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
+package com.rakovets.course.java.core.practice.oop_principles.Geometry;
+
+import com.rakovets.course.java.core.util.NumberUtil;
+
+public class Foursquare extends Rectangle {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Foursquare.java:6]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Height.java [hunk 15]
```diff
@@ -0,0 +1,6 @@
+package com.rakovets.course.java.core.practice.oop_principles.Geometry;
+
+public interface Height {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/Height.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/ShapeUtils.java [hunk 16]
```diff
@@ -0,0 +1,12 @@
+package com.rakovets.course.java.core.practice.oop_principles.Geometry;
+
+public class ShapeUtils {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/Geometry/ShapeUtils.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/BattleGroundTest/ArcherTest.java [hunk 17]
```diff
@@ -0,0 +1,79 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGroundTest;
+
+import com.rakovets.course.java.core.practice.oop_principles.BattleGround.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ArcherTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/BattleGroundTest/ArcherTest.java:8]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/BattleGroundTest/MagTest.java [hunk 18]
```diff
@@ -0,0 +1,87 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGroundTest;
+
+import com.rakovets.course.java.core.practice.oop_principles.BattleGround.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class MagTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/BattleGroundTest/MagTest.java:8]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/BattleGroundTest/WitchTest.java [hunk 19]
```diff
@@ -0,0 +1,85 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGroundTest;
+
+import com.rakovets.course.java.core.practice.oop_principles.BattleGround.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class WitchTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/BattleGroundTest/WitchTest.java:8]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/EllipseTest.java [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Ellipse;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Figure;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Rectangle;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Triangle;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class EllipseTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/EllipseTest.java:11]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/EquilateralTriangleTest.java [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.EquilateralTriangle;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Figure;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Height;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Rectangle;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class EquilateralTriangleTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/EquilateralTriangleTest.java:11]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/FoursquareTest.java [hunk 22]
```diff
@@ -0,0 +1,71 @@
+package com.rakovets.course.java.core.practice.oop_principles.GeometryTest;
+
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class FoursquareTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/FoursquareTest.java:8]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/RectangleTest.java [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Diagonal;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Figure;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Foursquare;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Rectangle;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class RectangleTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/RectangleTest.java:11]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/ShapeUtilsTest.java [hunk 24]
```diff
@@ -0,0 +1,44 @@
+package com.rakovets.course.java.core.practice.oop_principles.GeometryTest;
+
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ShapeUtilsTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/ShapeUtilsTest.java:8]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/TriangleTest.java [hunk 25]
```diff
@@ -0,0 +1,80 @@
...
+
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.EquilateralTriangle;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Figure;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Height;
+import com.rakovets.course.java.core.practice.oop_principles.Geometry.Triangle;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TriangleTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/GeometryTest/TriangleTest.java:11]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Mag.java [hunk 26]
```diff
@@ -0,0 +1,23 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public class Mag extends Hero {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Mag.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Mortal.java [hunk 27]
```diff
@@ -0,0 +1,6 @@
+package com.rakovets.course.java.core.practice.oop_principles.BattleGround;
+
+public interface Mortal {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/BattleGround/Mortal.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

**Comment by AlenaTrafimovich** [comment]
Corrected comments

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**